release/default.nix

```diff
@@ -0,0 +1,20 @@
+# SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+{ pkgs ? import (import ../nix/sources.nix).nixpkgs { } }:
+let
+  executable = (import ../crossref-verifier.nix {
+    static = true;
+  }).components.exes.crossref-verify;
+  joinByBasename = name: paths:
+    pkgs.runCommandNoCC name { } (''
+      mkdir $out
+    '' + pkgs.lib.concatMapStrings (path: ''
+      ln -s ${path} $out/${builtins.baseNameOf path}
+    '') paths);
+in joinByBasename "crossref-verifier-release" [
+  "${executable}/bin"
+  ../LICENSES
```

Comment:
Why do you include the licences folder in the release?

Reply:
Because licensing information is not included in the binary I don't think and it's better to include it in the distribution for legal reasons.

Reply:
I'm bad with legal stuff, can you explain the reasons?

Reply:
```
   3. Responsibilities

      3.1. Distribution of Source Form

All distribution of Covered Software in Source Code Form, including any Modifications
that You create or to which You contribute, must be under the terms of this
License. You must inform recipients that the Source Code Form of the Covered
Software is governed by the terms of this License, and how they can obtain
a copy of this License. You may not attempt to alter or restrict the recipients'
rights in the Source Code Form.

      3.2. Distribution of Executable Form

      If You distribute Covered Software in Executable Form then:

(a) such Covered Software must also be made available in Source Code Form,
as described in Section 3.1, and You must inform recipients of the Executable
Form how they can obtain a copy of such Source Code Form by reasonable means
in a timely manner, at a charge no more than the cost of distribution to the
recipient; and

(b) You may distribute such Executable Form under the terms of this License,
or sublicense it under different terms, provided that the license for the
Executable Form does not attempt to limit or alter the recipients' rights
in the Source Code Form under this License.
```

I'm definitely no lawyer either, but it appears to me that distributing the license with the executable is a good idea even if not required.

Reply:
Ok

Reply:
GitHub seems to put source code into each release by default (there is `Source code` in assets [here](https://github.com/serokell/crossref-verifier/releases/tag/auto-release)). And AFAIU for distribution in Executable Form we only have to make it easy to obtain Source Code.

So:
1. Licenses are not necessary currently because if a person goes to GitHub releases to download this software in Executable Form the will clearly see how to download this software in Source Form.
2. If we ever want to distribute `crossref-verifier` somewhere else (outside of GitHub releases), including only the license is not sufficient, we should also include Source Code or a link to it (or somehow else tell people how to obtain Source Code).

At the same time I see that we use a thing with MIT license and it says that we can do whatever we want with that software, but:
> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

So apparently that license requires us to include at least the MIT license into distribution. I don't know whether `release.tar.gz` is considered as its own "distribution" or only the whole GitHub release can be considered as a distribution. In the latter case including the license seems to be unnecessary because it is part of "Source code".

Without further investigation I agree that including all licenses is safer anyway, but I'd like to get answers, will try asking in #legal.

Reply:
Yeah, better be safe than sorry and 28kB (~7kB compressed) is not a big deal compared to the size of the executable anyways.

Reply:
Ok, let's include the license for safety. But are you sure we need the whole LICENSES and not the LICENSE file?

Reply:
Well, different licenses apply to different source files. In fact, the `LICENSE` file is present primarily because GitHub recognizes it and shows on the main repo page.

But it doesn't mean that all source files are covered by only this license. E. g. `sources.nix` belongs to nmattia and there the MIT license applies and it requires to put its copyright notice into anything you distribute. And `source.nix` participates in building the archive.

Reply:
Ok, I'm clearly far from being a lawyer, let's keep it like that. But out of interest, I wonder what would be different if we imported the `sources.nix` file from an external url without including it into the repository, like we do with the libraries that we fetch from git. Would we not have to include the copyright notice then?

Reply:
I think no, but I don't have any explanation. Feel free to ask in #legal.